src/vmm/src/persist.rs

```diff
@@ -204,6 +220,10 @@ fn snapshot_state_to_file(
     // Translate the microVM version to its corresponding snapshot data format.
     let snapshot_data_version = match version {
         Some(version) => match FC_VERSION_TO_SNAP_VERSION.get(version) {
+            Some(&FC_V0_23_SNAP_VERSION) => {
+                validate_devices_number(device_manager.used_irqs_count())?;
```

Comment:
Also, if you do it right there you would have CreateSnapshotParams.version which is already a string, so you would not need to define FC_V0_23_SNAP_VERSION.

Reply:
I see your point in limiting the function's responsibility to only saving the microVM state. However, I was thinking that it would make for a more readable code to perform all snapshot version handling in the same place. WDYT?

Reply:
I think we can keep it as is for now, but in the future we need separate the version handling as it will get beefy in time.

Reply:
Agreed.